### PR TITLE
wayland: Support HiDPI for cursor

### PR DESCRIPTION
**Pull request by Nefsen402**

### CODE REVIEW

**Comment by Nefsen402** [comment]
!393 Seems to do the same thing, plus some extra.

File: wayland.c [hunk 1]
```diff
@@ -176,6 +176,42 @@ static void touch_handle_up(void *data, struct wl_touch *wl_touch,
 	seat->touch.pts[id].surface = NULL;
 }
 
+void load_cursor(struct mako_state *state, uint32_t scale) {
+	const char *cursor_name = "left_ptr";
+
+	//don't reload the cursor if what we have already can be used
+	if (state->cursor.theme != NULL && state->cursor.scale == scale)
```
Comment (emersion) [inline wayland.c:183]: Style: braces are mandatory. Applies to the whole patch.

File: wayland.c [hunk 2]
```diff
@@ -176,6 +176,42 @@ static void touch_handle_up(void *data, struct wl_touch *wl_touch,
 	seat->touch.pts[id].surface = NULL;
 }
 
+void load_cursor(struct mako_state *state, uint32_t scale) {
```
Comment (emersion) [inline wayland.c:179]: Nit: this function can be `static`.

File: wayland.c [hunk 3]
```diff
@@ -186,10 +222,20 @@ static void pointer_handle_enter(void *data, struct wl_pointer *wl_pointer,
 	seat->pointer.y = wl_fixed_to_int(surface_y);
 	seat->pointer.surface = get_surface(state, wl_surface);
 
+	int scale = 1;
+
+	struct mako_surface *surface;
+	wl_list_for_each(surface, &state->surfaces, link) {
+		if (surface->surface_output->scale > scale) scale = surface->surface_output->scale;
```
Comment (emersion) [inline wayland.c:229]: Style: braces are mandatory

**Review by emersion — CHANGES_REQUESTED** [review]
Thanks for the patch! Here are some minor comments. The meat of the patch looks good.

**Review by emersion — APPROVED** [review]
Thanks!

**Pull request merged**